app/hooks/useTaco.ts

```diff
@@ -0,0 +1,101 @@
+import { SiweMessage } from "@didtools/cacao";
+import {
+  conditions,
+  decrypt,
+  encrypt,
+  ThresholdMessageKit,
+} from "@nucypher/taco";
+import {
+  SingleSignOnEIP4361AuthProvider,
+  USER_ADDRESS_PARAM_EXTERNAL_EIP4361,
+} from "@nucypher/taco-auth";
+import { DIDSession } from "did-session";
+import { ethers } from "ethers";
+import { useContext } from "react";
+import { TACoContext } from "../context/TACoContext";
+
+export default function useTaco() {
+  const { isInitialized, ritualId, domain } = useContext(TACoContext);
+
+  async function encryptWithTACo(
+    messageToEncrypt: string,
+    condition: conditions.condition.Condition,
+    provider: ethers.providers.Provider,
+    signer: ethers.Signer,
+  ) {
+    if (!isInitialized) return;
+
+    const tmk = await encrypt(
+      provider,
+      domain,
+      messageToEncrypt,
+      condition,
+      ritualId,
+      signer,
+    );
+
+    return encodeB64(tmk.toBytes());
+  }
+
+  async function decryptWithTACo(
+    encryptedMessage: string,
+    provider: ethers.providers.Provider,
+  ) {
+    if (!isInitialized) return;
+
+    const siweInfo = await loadSiweFromOrbisSession();
+    if (!siweInfo.message || !siweInfo.signature) {
+      console.error("SIWE message or signature missing");
+      return "<Decryption failed>";
```

Comment:
Perhaps something to indicate different failure mode?

```suggestion
      return "<User not already logged in>";
```

Reply:
It's a more comprehensible string indeed. The problem that I see is that, if I'm understanding well, the lack of `siweInfo.message` or `siweInfo.signature` can come from an error when the [async function `DIDSession.fromSession(session)` is called](https://github.com/nucypher/taco-orbisdb-demo/blob/ee218bc87b623a17dd97c2ce03cce1d06556ee2f/app/hooks/useTaco.ts#L85-L86). 

This function eventually will call this function on `dids` package and it seems that there are multiple causes that can lead an error that is not related with the "User not already logged in": https://github.com/ceramicnetwork/js-did/blob/def0a5fed152a42ffeef87bafb4619816f9954f6/packages/dids/src/did.ts#L248

Does it make sense to you?

Reply:
```suggestion
      return "<Decryption failed. This may be because the user has not yet logged in.>";
```